FlowCrypt/Controllers/Compose/ComposeViewController.swift

```diff
@@ -996,6 +1019,38 @@ extension ComposeViewController {
             }
         }
     }
+
+    private func setMessagePassword() {
+        Task {
+            contextToSend.password = await enterMessagePassword()
+            node.reloadSections([Section.password.rawValue], with: .automatic)
+        }
+    }
+
+    private func enterMessagePassword() async -> String? {
```

Comment:
I like modals as async methods - it really cleans up the logic of wherever it ends up used.

Reply:
Yeah, `async/await` is a great addition to Swift, makes code much more readable.
Especially in such cases as modals where previously we needed several action handlers.